src/core/mount.c

```diff
@@ -1622,6 +1622,12 @@ static int mount_setup_existing_unit(
 
         MOUNT(u)->from_proc_self_mountinfo = true;
 
+        /* when we have both info from /etc/fstab and from /proc/self/mountinfo, and the mount generated by
+         * fstab is in a failed state then remove any BindsTo= deps generated due to fstab, and instead
+         * snthesize BindsTo= on the block device listed in /proc/self/mountinfo. */
+        if (u->fragment_path && IN_SET(MOUNT(u)->state, MOUNT_DEAD, MOUNT_FAILED))
```

Comment:
I don't understand why the condition is like this here. It doesn't match what @poettering wrote in https://github.com/systemd/systemd/pull/15389#issuecomment-629514514, and it doesn't seem right: the removal of dependencies generated based on /etc/fstab shouldn't be conditional. So I think this here should be just
```
    unit_remove_dependencies(u, UNIT_DEPENDENCY_FILE);
```
(no `if`),